crcmocks/rbac.py

```diff
@@ -1,33 +1,72 @@
+from copy import deepcopy
+
 from flask import jsonify
 from flask import Blueprint
+from flask import request
+
+from crcmocks.util import get_user_rh_identity
+from crcmocks.config import DEFAULT_PERMISSIONS
 
 
 blueprint = Blueprint("rbac", __name__)
 
-rbac_response = {
-    "meta": {"count": 30, "limit": 1000, "offset": 0},
+BASE_RBAC_RESPONSE = {
+    "meta": {"count": None, "limit": None, "offset": 0},
     "links": {
         "first": "/api/rbac/v1/access/?application=&format=json&limit=1000&offset=0",
         "next": None,
         "previous": None,
         "last": "/api/rbac/v1/access/?application=&format=json&limit=1000&offset=0",
     },
-    "data": [
-        {"resourceDefinitions": [], "permission": "advisor:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-    ],
+    "data": [],
 }
 
 
-# TODO: allow customizable rbac response per user
-@blueprint.route("/v1/access/")
+# TODO: support resource definitions?
+@blueprint.route("/v1/access/", methods=["GET"])
 def rbac_access():
+    application = request.args.get("application")
+    limit = int(request.args.get("limit", 100))
+    # get the identity header
+    identity_header = request.headers.get("X-Rh-Identity")
+
+    if not identity_header:
+        return "No identity header present in request\n", 401
+
+    # get the user
+    user, username, account_number = get_user_rh_identity(identity_header)
+
+    if not user:
+        return f"No user found in TinyDB with username:" \
+               f" {username} or account_number: {account_number}\n", 404
+    if len(user) > 1:
+        return f"Multiple users found with:" \
+               f" {username} or account_number: {account_number}\n", 406
+
+    # get the user permissions
+    if user[0].get("entitlements") is not None:
+        permissions = user[0]["permissions"].split(",")
+    else:
+        permissions = DEFAULT_PERMISSIONS.copy()
+
+    rbac_response = deepcopy(BASE_RBAC_RESPONSE)
+    rbac_response["meta"]["count"] = len(permissions)
+    rbac_response["meta"]["limit"] = limit
+
+    count = 1
```

Comment:
I think you need to start with `0` here, if permissions is an empty list you'll end up with a count of '1'. If permissions is a list of 1 item, you'd end up with count of '2'

Reply:
Ah @bsquizz this count is the actual count of items, it's just used to set the limit, the actual count is the `len(permission)` line.

Reply:
Right but `len(permission)` is not an accurate count if you filtered by `application`

For ex if you had 10 permissions, only 1 advisor permission, and filtered by `application=advisor`, count should only be `1` right?

Reply:
Yup, I realized that after I made that comment and moved the `len` to after the permissions, I'll just change it to limit_count since it's the same now.